src/common/utils/url.ts

```diff
@@ -178,24 +136,26 @@ export function equalUrlsIgnoringSubpath(url1: URL, url2: URL, ignoreScheme?: bo
     return url1.origin.toLowerCase() === url2.origin.toLowerCase();
 }
 
-function isTrustedURL(url: URL | string, teams: TeamWithTabs[]) {
+function isTrustedURL(url: URL | string, rootURL: URL | string) {
     const parsedURL = parseURL(url);
-    if (!parsedURL) {
+    const rootParsedURL = parseURL(rootURL);
+    if (!parsedURL || !rootParsedURL) {
         return false;
     }
-    return getView(parsedURL, teams) !== null;
+    return (getFormattedPathName(rootParsedURL.pathname) !== '/' && equalUrlsWithSubpath(rootParsedURL, parsedURL)) ||
```

Comment:
This check is here to ensure that whatever URL we are allowing is "trusted", meaning that it belongs to one of the configured servers. We were using `getView()` for this, but there was never a case where we needed to check across servers or tabs, so this is enough.

Reply:
It's usually enough, but we want to make sure that the server URL matches exactly. It's unlikely, but I could see a case where an MM instance is hosted on a potentially untrusted domain that has a trusted subpath.